pymongo/mongo_client.py

```diff
@@ -2125,3 +2138,23 @@ def __enter__(self):
 
     def __exit__(self, exc_type, exc_val, exc_tb):
         return self.handle(exc_type, exc_val)
+
+
+def _after_fork():
+    """
+    Reinitialiazes the locks in child process and resets the
+    topologies in all MongoClients.
+    """
+    # Perform cleanup in clients (i.e. get rid of topology)
+    for client in MongoClient._clients:
+        client._after_fork()
+
+    # Reinitialize locks
+    MongoClientLock._reset_locks()
+
+    # These can't use MongoClientLock due to circular imports.
+    ObjectId._reset_lock()
```

Comment:
If an app _only_ uses bson and never imports pymongo, we still need ObjectId to be fork() safe. Let's move the ObjectId logic to the bson module.